oeps/oep-0008.rst

```diff
@@ -0,0 +1,323 @@
+| `Replaced-By` |                                           |
++---------------+-------------------------------------------+
+
+Abstract
+========
+
+The edX ecosystem has grown from a handful of developers working on a few
+repos to many developers working on many repos. As such, questions arise as to
+tools, frameworks, and patterns to use in unit testing. This proposal is
+intended to codify best practices and to provide a reference for contributors
+when making choices.
+
+
+Motivation
+==========
+
+Having common unit testing best practices across the organization's repo will
+increase the productivity of developers by reducing the ramp-up cost when
+moving among repos. It will also increase the quality of the software by
+decreasing the cognitive load on developers who must call to mind the various
+tips and techniques while using the disparate tools and frameworks.
+
+
+Specification
+=============
+
+CI (Continuous Integration) Framework
+-------------------------------------
+We use `Travis CI`_ for both public and private repos, with the exception of
+the edx-platform repo for which we use an internally-managed Jenkins
+environment.
+
+Guidelines:
+^^^^^^^^^^^
+
+* Configuration file: ``.travis.yml``
+
+* Use the container-base (sudo-less) CI environment, if possible
+
+* Execute tests for each supported Python version, and if appropriate also in
+  combination with different Django versions
+
+* Configure with automated deployment of tagged releases to `PyPI`_
+
+
+Task Management
+---------------
+We use Makefiles for task management. The reason for this choice is that
+almost any system no matter the OS should be able to use them without any
+special software installation or configuration.
+
+If the Makefile gets too complex, use it as a wrapper to `Paver`_ scripts,
+which are written in Python.
+
+Standard Makefile Targets:
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
++--------------+--------------------------------------------------------------+
+| target       | description                                                  |
++==============+==============================================================+
+| help         | display a help message                                       |
++--------------+--------------------------------------------------------------+
+| pip-compile  | update the ``requirements/*.txt`` files with the             |
+|              | latest packages satisfying ``requirements/*.in``             |
++--------------+--------------------------------------------------------------+
+| requirements | install development environment requirements                 |
++--------------+--------------------------------------------------------------+
+| quality      | check coding style                                           |
++--------------+--------------------------------------------------------------+
+| test         | run all unit, integration, and acceptance tests for the repo |
```

Comment:
I think having a way to run tests w/o quality, and then having a separate task that checks both, is a good way to go, since you often (as a developer) want to do one or the other, but not both.